File: 100/154 - Find Minimum in Rotated Sorted Array II.c
int findMin(int* nums, int numsSize) {
    int from = 0, to = numsSize - 1, mid;
    
    if (! numsSize)
        return 0;
    
    while (from < to) {
        mid = (from + to) / 2;
        if (nums[mid] < nums[to])
            to = mid;
        else if (nums[mid] > nums[to])
            from = mid + 1;
        else
            --to;
    }

    return nums[from];
}

File: csharp/154.find-minimum-in-rotated-sorted-array-ii.cs
/*
 * @lc app=leetcode id=154 lang=csharp
 *
 * [154] Find Minimum in Rotated Sorted Array II
 */
public class Solution {
    public int FindMin(int[] nums) {
        int left = 0, right = nums.Length - 1;
        while (left < right) {
            int mid = (left + right) / 2;
            if (nums[mid] < nums[right]) {
                right = mid;
            } else if (nums[mid] > nums[right]) {
                left = mid + 1;
            } else {
                --right;
            }
        }
        return nums[left];
    }
}

